ccinput/packages/psi4.py

```diff
@@ -0,0 +1,126 @@
+import basis_set_exchange
+import numpy as np
+
+from ccinput.utilities import (
+    get_method,
+    get_solvent,
+    get_basis_set,
+    clean_xyz,
+    get_distance,
+    get_angle,
+    get_dihedral,
+    get_npxyz,
+)
+from ccinput.calculation import Calculation
+from ccinput.constants import CalcType, ATOMIC_NUMBER, LOWERCASE_ATOMIC_SYMBOLS
+from ccinput.exceptions import InvalidParameter, ImpossibleCalculation
+
+
+class Psi4Calculation:
+    TEMPLATE = """
+molecule {{
+{charge} {multiplicity}  
+{coordinates}
+}}
+
+{command_line}
+"""
+
+    CALC_TYPES = {
+        CalcType.SP: ["sp","energy"],
+        CalcType.OPT: ["optimize"],
+        # CalcType.CONSTR_OPT,
+        # CalcType.FREQ,
+        # CalcType.TS,
+        # CalcType.OPTFREQ,
+    }
+        
+    def __init__(self, calc: Calculation):
+        self.calc = calc
+        self.has_scan = False
+        self.appendix = []
+        self.command_line = ""
+
+        self.commands = {}
+        self.solvation_radii = {}
+
+        self.confirmed_specifications = ""
+        self.xyz_structure = ""
+        self.input_file = ""
+
+        if self.calc.type not in self.CALC_TYPES:
+            raise ImpossibleCalculation(
```

Comment:
This should be `UnimplementedError` in order to keep everything consistent.

Reply:
Done.